src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -0,0 +1,1386 @@
+     * @param formatId Metadata namespace
+     * @return Actual path to object
+     * @throws IllegalArgumentException If entity is not object or metadata
+     * @throws NoSuchAlgorithmException If store algorithm is not supported
+     */
+    protected Path getRealPath(String pid, String entity, String formatId)
+            throws IllegalArgumentException, NoSuchAlgorithmException {
+        Path realPath;
+        if (entity.equalsIgnoreCase("object")) {
+            String objectCid = this.getPidHexDigest(pid, this.OBJECT_STORE_ALGORITHM);
+            String objShardString = this.getHierarchicalPathString(this.DIRECTORY_DEPTH, this.DIRECTORY_WIDTH,
+                    objectCid);
+            realPath = this.OBJECT_STORE_DIRECTORY.resolve(objShardString);
+
+        } else if (entity.equalsIgnoreCase("metadata")) {
+            String objectCid = this.getPidHexDigest(pid + formatId, this.OBJECT_STORE_ALGORITHM);
+            String objShardString = this.getHierarchicalPathString(this.DIRECTORY_DEPTH, this.DIRECTORY_WIDTH,
+                    objectCid);
+            realPath = this.METADATA_STORE_DIRECTORY.resolve(objShardString);
+
+        } else {
+            throw new IllegalArgumentException("FileHashStore.getRealPath - entity must be 'object' or 'metadata'");
+
+        }
+        return realPath;
+    }
+
+    /**
+     * Deletes a given object and its parent directories if they are empty
+     * 
+     * @param objectAbsPath Path of the object to delete
+     * @param pid           Authority-based identifier
+     * @param method        Calling method
+     * @throws IOException I/O error when deleting object or accessing directories
+     */
+    private void deleteObjectAndParentDirectories(Path objectAbsPath, String pid, String method) throws IOException {
+        // Delete file
+        Files.delete(objectAbsPath);
+
+        // Then delete any empty directories
+        Path parent = objectAbsPath.getParent();
+        while (parent != null && this.isDirectoryEmpty(parent)) {
+            if (parent.equals(this.METADATA_STORE_DIRECTORY)) {
+                // Do not delete the metadata store directory
+                break;
+
+            } else {
+                Files.delete(parent);
+                logFileHashStore
+                        .debug("FileHashStore.deleteObjectAndParentDirectories - " + method
+                                + " : Deleting parent directory for: " + pid + " with parent address: " + parent);
+                parent = parent.getParent();
+
+            }
+        }
+    }
+
+    /**
+     * Checks whether a directory is empty or contains files. If a file is found, it
+     * returns true.
+     *
+     * @param directory Directory to check
+     * @return True if a file is found or the directory is empty, False otherwise
+     * @throws IOException If I/O occurs when accessing directory
+     */
+    private boolean isDirectoryEmpty(Path directory) throws IOException {
```

Comment:
you could re-use this method on line 170, instead of having 2 different ways to do the same thing

Reply:
I've re-used this method, thank you!